pkg/apply/desiredset_process_test.go

```diff
@@ -0,0 +1,93 @@
+package apply
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic/fake"
+	k8stesting "k8s.io/client-go/testing"
+)
+
+func Test_multiNamespaceList(t *testing.T) {
+	results := map[string]*unstructured.UnstructuredList{
+		"ns1": {Items: []unstructured.Unstructured{
+			{Object: map[string]interface{}{"name": "o1", "namespace": "ns1"}},
+			{Object: map[string]interface{}{"name": "o2", "namespace": "ns1"}},
+			{Object: map[string]interface{}{"name": "o3", "namespace": "ns1"}},
+		}},
+		"ns2": {Items: []unstructured.Unstructured{
+			{Object: map[string]interface{}{"name": "o4", "namespace": "ns2"}},
+			{Object: map[string]interface{}{"name": "o5", "namespace": "ns2"}},
+		}},
+		"ns3": {Items: []unstructured.Unstructured{}},
+	}
+
+	baseClient := fake.NewSimpleDynamicClient(runtime.NewScheme())
+	baseClient.PrependReactor("list", "*", func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
+		if strings.Contains(action.GetNamespace(), "error") {
+			return true, nil, errors.New("simulated failure")
+		}
+
+		return true, results[action.GetNamespace()], nil
+	})
+
+	// no namespaces
+	{
```

Comment:
In `pkg/objectset/objectset_test.go` we use "more native" approach and do table-driven tests by defining `tests` slice and iterating over it. This produces better `go test` output.
I believe it should be possible to rewrite this test with the same approach or am I missing a reason they are done this way here?

Reply:
When I initially started theses tests it seemed like I would need to call things between cases but looking now I can use that same pattern.  Ill make them consistent.